cassiopeia/solver/VanillaGreedySolver.py

```diff
@@ -0,0 +1,165 @@
+"""
+This file stores a subclass of GreedySolver, the VanillaGreedySolver. The
+inference procedure here is the "vanilla" Cassiopeia-Greedy, originally proposed
+in Jones et al, Genome Biology (2020). In essence, the algorithm proceeds by
+recursively splitting samples into mutually exclusive groups based on the
+presence, or absence, of the most frequently occurring mutation.
+"""
+import numpy as np
+import pandas as pd
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+from cassiopeia.solver import GreedySolver
+from cassiopeia.solver import solver_utilities as utils
+
+
+class VanillaGreedySolver(GreedySolver.GreedySolver):
+    def __init__(
+        self,
+        character_matrix: pd.DataFrame,
+        missing_char: str,
+        missing_data_classifier: Union[Callable, str],
+        meta_data: Optional[pd.DataFrame] = None,
+        priors: Optional[Dict] = None,
+        fuzzy_solver: bool = False,
+    ):
+
+        super().__init__(character_matrix, missing_char, meta_data, priors)
+
+        self.missing_data_classifier = missing_data_classifier
+        self.fuzzy_solver = fuzzy_solver
+
+    def perform_split(
+        self,
+        samples: List[int] = None,
+    ) -> Tuple[List[int], List[int]]:
+        """Performs a partition based on the most frequent (character, state) pair.
+
+        Uses the (character, state) pair to split the list of samples into
+        two partitions. In doing so, the procedure makes use of the missing
+        data classifier passed into the class.
+
+        Args:
+            samples: A list of samples to partition
+
+        Returns:
+            A tuple of lists, representing the left and right partitions
+        """
+        if not samples:
+            samples = list(range(self.prune_cm.shape[0]))
+        F = utils.compute_mutation_frequencies(
+            self.prune_cm, self.missing_char, samples
```

Comment:
The perform_split call in the abstract class's implementation of `solve()` passes `F` as an argument. Why does this function not take in `F` as an argument, and why does it recompute it here?